spec/services/authn_context_resolver_spec.rb

```diff
@@ -291,42 +291,139 @@
       end
     end
 
-    context 'IAL2 service provider' do
-      it 'uses the IAL ACR if one is present' do
-        service_provider = build(:service_provider, ial: 2)
-
-        acr_values = [
-          'http://idmanagement.gov/ns/assurance/aal/1',
-          'http://idmanagement.gov/ns/assurance/ial/1',
-        ].join(' ')
-
-        result = AuthnContextResolver.new(
+    context 'with an IAL2 service provider' do
+      let(:service_provider) { build(:service_provider, ial: 2) }
+      subject do
+        AuthnContextResolver.new(
           user: user,
           service_provider: service_provider,
           vtr: nil,
           acr_values: acr_values,
-        ).resolve
-
-        expect(result.identity_proofing?).to eq(false)
-        expect(result.aal2?).to eq(false)
+        )
+      end
+      let(:result) { subject.resolve }
+
+      context 'if IAL ACR value is present' do
+        let(:acr_values) do
+          [
+            'http://idmanagement.gov/ns/assurance/ial/1',
+            'http://idmanagement.gov/ns/assurance/aal/1',
+          ].join(' ')
+        end
+
+        it 'uses the IAL ACR if one is present' do
+          expect(result.identity_proofing?).to be false
+          expect(result.aal2?).to be false
```

Comment:
I find this expectation surprising. Don't we only support aal2's? Requesting no AAL results in an aal1?

Reply:
we require aal2 for identity proofing. for ial1 we only require MFA, which is not aal2. https://developers.login.gov/oidc/authorization/#aal_values